### PR TITLE
require fetchTimestamp but with timetolive opt-out

### PR DESCRIPTION
**Pull request by aappling-usgs**
Addresses issues #192 and #102, which are closely related

* require absence of `refetch` and `fetchTimestamp` in viz.yaml fetch item chunks
* require presence of `fetchTimestamp.xx` method for every fetcher `xx`
* built-in `fetchTimestamp.file` method writes a timestamp file at least once
* slightly easier-to-use `writeTimestamp` function for custom `fetchTimestamp` methods
* implemented a timetolive option for every fetch file in a new preferences.yaml file that users can put in their home project directory (and should not git commit, because this file is for _local_ preferences)
* scenarios should work as in the table at https://github.com/USGS-VIZLAB/vizlab/issues/192#issuecomment-329681645

### CODE REVIEW

File: R/createMakefiles.R [hunk 1]
```diff
@@ -329,23 +334,43 @@ createMakeItem.fetch <- function(item.info, ...) {
...
+    phony.timestamp <- createMakeBatchRule(
       target=timestamp.id,
       fun='fetchTimestamp',
       funargs=c(viz=squote(item.info$id)),
       scripts=item.info$scripts,
       logfile=paste0('fetch/', timestamp.id, '.Rout'))
+    # interleave a gnu make conditional (I hope it works on all systems!!) for exceedance of timetolive
+    phony.timestamp.split <- strsplit(phony.timestamp, split='\n')[[1]]
+    phony.timestamp.full <- c(phony.timestamp.split[1],
+      sprintf("ifeq ($(shell echo $(shell Rscript -e \"vizlab::exceededTimeToLive('%s')\" 2> null)),TRUE)", item.info$id),
```
Comment (aappling-usgs) [inline R/createMakefiles.R]: whoops, just realized there's now a file named `null` in my repo, thought I was sending to empty. i probably meant `/dev/null`

Comment (ldecicco-USGS) [reply R/createMakefiles.R]: while you are fixing that, don't forget to add `importFrom("utils", "getS3method")` or however you want to do that to get Travis happy

Comment (aappling-usgs) [reply R/createMakefiles.R]: :+1:

**Comment by ldecicco-USGS** [comment]
FYI, it *looks* pretty awesome, if you are happy with it, I'd vote merging and if we discover issues make new issues. We'll want to merge this and https://github.com/USGS-VIZLAB/example/pull/18

**Pull request merged**